testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/scripts/ScriptUtil.java

```diff
@@ -0,0 +1,186 @@
+
+    public String JBOSS_HOME;
+    public String TEST_DIR;
+    public String SECURITY_POLICY;
+    public String WSPROVIDE_INPUT_DIR;
+    public String ABSOUTPUT;
+
+    @Before
+    public void setup() throws Exception {
+        JBOSS_HOME = System.getProperty("jboss.dist");
+        ENDPOINT_CLASS = "org.jboss.as.testsuite.integration.scripts.test.tools.Echo1Impl";
+        TEST_DIR = getTestResourcesDirAsFile().getAbsolutePath();
+        WSPROVIDE_INPUT_DIR = TEST_DIR + FS + "ws" + FS + "scripts";
+        SECURITY_POLICY = new File(WSPROVIDE_INPUT_DIR + FS + "Echo1-security.policy").getAbsolutePath();
+        ABSOUTPUT = new File(TEST_DIR).getAbsolutePath();
+    }
+
+    /**
+     * Execute <b>command</b> in separate process, copy process input to <b>os</b>. If process will fail, display custom <b>message</b> in assertion.
+     * @param command command to execute
+     * @param os output stream to copy process input to. If null, <b>System.out</b> will be used
+     * @param message message to display if assertion fails
+     * @param env environment
+     * @throws IOException if I/O error occurs
+     */
+    public static String executeCommand(String command, OutputStream os,
+                                        String message, Map<String, String> env) throws IOException, InterruptedException {
+        if (command == null)
+            throw new NullPointerException( "Command cannot be null" );
+
+        log.info("Executing command: " + command);
+
+        StringTokenizer st = new StringTokenizer(command, " \t\r");
+        List<String> tokenizedCommand = new LinkedList<String>();
+        while (st.hasMoreTokens()) {
+            // PRECONDITION: command doesn't contain whitespaces in the paths
+            tokenizedCommand.add(st.nextToken());
+        }
+
+        String result = "";
+        try {
+            result = executeCommand(tokenizedCommand, os, message, env);
+        }
+        catch (IOException e) {
+            log.warn("Make sure there are no whitespaces in command paths", e);
+            throw e;
+        }
+        return result;
+    }
+
+    /**
+     * Execute <b>command</b> in separate process, copy process input to <b>os</b>. If process will fail, display custom <b>message</b> in assertion.
+     * @param command command to execute
+     * @param os output stream to copy process input to. If null, <b>System.out</b> will be used
+     * @param message message to display if assertion fails
+     * @param env environment
+     * @throws IOException if I/O error occurs
+     */
+    private static String executeCommand(List<String> command, OutputStream os,
+                                         String message, Map<String, String> env) throws IOException, InterruptedException {
+
+        ProcessBuilder pb = new ProcessBuilder(command);
+        if (env != null) {
+            for (String variable : env.keySet()) {
+                pb.environment().put(variable, env.get(variable));
```

Comment:
Regarding the test failures - maybe the M2_HOME environment variable isn't getting to the child process?  Or, maybe there are multiple (or no) Maven settings files on the server?  @ctomc do you know how we're presently configuring Maven settings on the server?

Reply:
It's essentially failing to find xalan:serializer:2.7.1.jbossorg-4, which I can see exists at https://repository.jboss.org/nexus/service/local/repositories/thirdparty-releases/content/xalan/serializer/2.7.1.jbossorg-4/serializer-2.7.1.jbossorg-4.jar - hence why I think it's some kind of Maven environment setup issue.

JBoss Modules searches the following places for settings, in order:

* The URL found in the ``jboss.modules.settings.xml.url`` system property
* The ``.m2/settings.xml`` file inside of the user home directory (from the system property ``user.home``)
* The ``conf/settings.xml`` file inside of the directory in the ``M2_HOME`` environment variable

If that fails, then a set of default settings is used - which would exclude ``repository.jboss.org``.

Reply:
we do use maven settings property everywhere in testsuite.
see https://github.com/wildfly/wildfly/blob/master/testsuite/integration/pom.xml#L275
this are the settings you are looking for.
or in short, you need to pass "maven.repo.local" system property.
Value you usually get from maven directly, ${settings.localRepository}

so just make sure that maven.repo.local is properly forwarded.